core/src/test/scala/unit/kafka/common/TopicTest.scala

```diff
@@ -81,12 +81,12 @@ class TopicTest {
     val periodFirstMiddleLastNone = List(".topic", "to.pic", "topic.", "topic")
     val underscoreFirstMiddleLastNone = List("_topic", "to_pic", "topic_", "topic")
```

Comment:
The test is failing because the last two topics in each list do not collide by the new definition.